### PR TITLE
Drop 0.6 support, and with it Compat

### PR DESCRIPTION
**Pull request by oxinabox**
This package is stable.
There were no new features added to this since August 2014
https://github.com/JuliaIO/IniFile.jl/commit/507cde1d7c6299582fb7d51e68f82d785c264400

The only changes have been for compatibility, and for infrastructure, and for docs.
As such, I predict that there will be no changes made going forward in the 0.7 timeframe, that we will wish to make available to 0.6 users.
Thus there is no reason to be using Compat to support both julia 0.6 and julia 0.7 with the same branch.

So this just drops Compat and julia 0.6 outright.
0.6 users can still enjoy IniFile.jl v0.4.0

(Motivation for this is Compat is broken for me right now. (EDIT: but I think that was due to an update failure or something on my part, still not required.))

### CODE REVIEW

File: REQUIRE [hunk 1]
```diff
@@ -1,2 +0,0 @@
-julia 0.6
```
Comment (aviks) [inline REQUIRE]: For some time, the registry will be populated via METADATA, so it makes sense to keep the REQUIRE with a `julia 0.7+`  lower bound

File: .travis.yml [hunk 2]
```diff
@@ -4,7 +4,6 @@ os:
   - linux
   - osx
 julia:
-  - 0.6
   - nightly
```
Comment (aviks) [inline .travis.yml]: I think `0.7` works on travis configs now?

Comment (oxinabox) [reply .travis.yml]: Let's find out

File: REQUIRE (hunk 1 above)
Comment (tkelman) [reply REQUIRE]: `0.7+` means only installable on versions greater than any release of 0.7.0, probably not what you actually want

Comment (oxinabox) [reply REQUIRE]: Right, fixed

```
julia> v"0.7.0-" < v"0.7.0-DEV" < v"0.7.0-alpha" < v"0.7.0" < v"0.7.0+"
true
```

**Pull request merged**